new resource: `azurerm_sentinel_data_connector_office_365` (#10671)
terraform-providers/terraform-provider-azurerm: website/docs/r/sentinel_data_connector_office_365.html.markdown

---
subcategory: "Sentinel"
layout: "azurerm"
page_title: "Azure Resource Manager: azurerm_sentinel_data_connector_office_365"
description: |-
  Manages a Office 365 Data Connector.
---

# azurerm_sentinel_data_connector_office_365

Manages a Office 365 Data Connector.

## Example Usage

```hcl
resource "azurerm_resource_group" "example" {
  name     = "example-rg"
  location = "west europe"
}

resource "azurerm_log_analytics_workspace" "example" {
  name                = "example-workspace"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  sku                 = "PerGB2018"
}

resource "azurerm_sentinel_data_connector_office_365" "example" {
  name                       = "example"
  log_analytics_workspace_id = azurerm_log_analytics_workspace.example.id
}
```

## Arguments Reference

The following arguments are supported:

> - `log_analytics_workspace_id` - (Required) The ID of the Log Analytics Workspace that this Office 365 Data Connector resides in. Changing this forces a new Office 365 Data Connector to be created.

Reviewer: ```suggestion
* `log_analytics_workspace_id` - (Required) The ID of the Log Analytics Workspace that this Office 365 Data Connector resides in. Changing this forces a new Office 365 Data Connector to be created.
```

* `name` - (Required) The name which should be used for this Office 365 Data Connector. Changing this forces a new Office 365 Data Connector to be created.

---

> - `exchange_enabled` - (Optional) Should the exchange data be enabled? Defaults to `true`.

Reviewer: presumably this should be:

```suggestion
* `exchange_enabled` - (Optional) Should the Exchange data connector be enabled? Defaults to `true`.
```

> - `share_point_enabled` - (Optional) Should the share point data be enabled? Defaults to `true`.

Reviewer: presumably this should be:

```suggestion
* `sharepoint_enabled` - (Optional) Should the SharePoint data connector be enabled? Defaults to `true`.
```

> - `teams_enabled` - (Optional) Should the teams data be enabled? Defaults to `true`.

Reviewer: ```suggestion
* `teams_enabled` - (Optional) Should the Microsoft Teams data connector be enabled? Defaults to `true`.
```

-> **NOTE**: At least one of `exchange_enabled`, `shared_point_enabled` and `teams_enabled` has to be specified.

> - `tenant_id` - (Optional) The ID of the tenant that this Office 365 Data Connector connects to. Changing this forces a new Office 365 Data Connector to be created.

Reviewer: ```suggestion
* `tenant_id` - (Optional) The ID of the tenant that this Office 365 Data Connector connects to. Changing this forces a new Office 365 Data Connector to be created.

-> **Note:** Terraform will use the Tenant ID for the current Subscription is this is unspecified.
```

## Attributes Reference

In addition to the Arguments listed above - the following Attributes are exported:

> - `id` - The ID of the Office 365 Data Connector.

Reviewer: ```suggestion
* `id` - The ID of the Office 365 Data Connector.
```

}

resource "azurerm_log_analytics_workspace" "example" {
  name                = "example-workspace"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  sku                 = "PerGB2018"
}

resource "azurerm_sentinel_data_connector_office_365" "example" {
  name                       = "example"
  log_analytics_workspace_id = azurerm_log_analytics_workspace.example.id
}
```

## Arguments Reference

The following arguments are supported:

* `log_analytics_workspace_id` - (Required) The ID of the Log Analytics Workspace that this Office 365 Data Connector resides in. Changing this forces a new Office 365 Data Connector to be created.

* `name` - (Required) The name which should be used for this Office 365 Data Connector. Changing this forces a new Office 365 Data Connector to be created.

---

* `exchange_enabled` - (Optional) Should the Exchange data connector be enabled? Defaults to `true`.

* `sharepoint_enabled` - (Optional) Should the SharePoint data connector be enabled? Defaults to `true`.

* `teams_enabled` - (Optional) Should the Microsoft Teams data connector be enabled? Defaults to `true`.

> -> **NOTE**: At least one of `exchange_enabled`, `shared_point_enabled` and `teams_enabled` has to be specified.

Reviewer: ```suggestion
-> **NOTE**: At least one of `exchange_enabled`, `sharepoint_enabled` and `teams_enabled` has to be specified.
```

> * `tenant_id` - (Optional) The ID of the tenant that this Office 365 Data Connector connects to. Changing this forces a new Office 365 Data Connector to be created.

Reviewer: ```suggestion
* `tenant_id` - (Optional) The ID of the Tenant that this Office 365 Data Connector connects to. Changing this forces a new Office 365 Data Connector to be created.
```
